cluster_agent/settings.py

```diff
@@ -54,6 +52,20 @@ class Settings(BaseSettings):
     # Single user submitter settings
     SINGLE_USER_SUBMITTER: Optional[str]
 
+    @validator("BASE_SLURMRESTD_URL", always=True)
+    def _validate_slurm_version_on_url(cls, v):
+        """
+        Add the default slurm version to the url if it is not present, for backward compatibility.
+        """
+        if not re.match(r".*/slurm/v\d+\.\d+\.\d+$", v):
```

Comment:
Why not make the environment setting "SLURM_RESTD_VERSION" and then build the URL from that?

Reply:
That was my first thought, but then I realized every request would use both `BASE_SLURMRESTD_URL` + `SLURM_RESTD_VERSION`, my second thought was why not keep it simple and provide them combined directly (no need to add extra setting to charms and bundles)? I'm ok with either way, what do you prefer?

Reply:
Well, I think you could make the `BASE_SLURMRESTD_URL` a "computed" setting that is set by the validator and make `SLURM_RESTD_VERSION` the setting that is actually used. There's some good examples of this in the Jobbergate CLI config: 

https://github.com/omnivector-solutions/jobbergate/blob/main/jobbergate-cli/jobbergate_cli/config.py#L67

Reply:
solved in [d5bc04a](https://github.com/omnivector-solutions/cluster-agent/pull/72/commits/d5bc04abab51bd04a63b624dcec28c1ced303b3d)